controllers/nodehealthcheck_controller_test.go

```diff
@@ -1959,6 +1959,101 @@ var _ = Describe("Node Health Check CR", func() {
+						g.Expect(getRemediationsCount(underTest)).To(Equal(3))
+					}, "5s", "1s").Should(Succeed())
+
+					// Phase 2: Make the forth node unhealthy - triggers first storm recovery
+					By("making one more node unhealthy - triggers storm recovery")
+					mockNodeGettingUnhealthy("healthy-worker-node-1")
+					// wait for node to turn unhealthy
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(4))
+					}, "15s", "100ms").Should(Succeed())
+
+					// Verify Storm recovery is activated
+					// 7 total, 3 healthy, 4 unhealthy, 3 remediations (additional remediation not created because of min healthy constraint)
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(3))
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(4))
+						g.Expect(getRemediationsCount(underTest)).To(Equal(3))
+						g.Expect(utils.IsConditionTrue(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
+						g.Expect(underTest.Status.StormRecoveryStartTime).ToNot(BeNil())
+					}, "5s", "100ms").Should(Succeed())
+
+					// Phase 3: Recover one node - storm recovery remains active due to 2-second delay
+					By("recovering one node - storm recovery remains active due to 2-second delay")
+					mockNodeGettingHealthy("unhealthy-worker-node-1")
+
+					//wait for node to recover
+					Eventually(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(4))
+					}, "5s", "100ms").Should(Succeed())
+					// 7 total, 4 healthy, 3 unhealthy, 2 remediations (one removed due to healthy node), 1 remediation is pending to be created (not created because of storm)
+					Consistently(func(g Gomega) {
+						g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+						g.Expect(*underTest.Status.HealthyNodes).To(Equal(4))
+						g.Expect(len(underTest.Status.UnhealthyNodes)).To(Equal(3))
+						g.Expect(getRemediationsCount(underTest)).To(Equal(2))
+						g.Expect(utils.IsConditionTrue(underTest.Status.Conditions, v1alpha1.ConditionTypeStormActive, v1alpha1.ConditionReasonStormThresholdChange)).To(BeTrue())
+					}, "2s", "100ms").Should(Succeed())
+					//expected termination time of the first storm
+					firstStormTerminationTime := underTest.Status.StormTerminationStartTime.Time.Add(underTest.Spec.StormTerminationDelay.Duration)
+					// Phase 4: wait for the delay to pass
+					time.Sleep(time.Millisecond * 1500)
```

Comment:
Why don't we just wait 2 seconds, as it defines at line 2783?

Reply:
Just saving a bit of time I guess. Presumably the previous validations (after mocking a node unhealthy) take some time as well.

Reply:
ok, but IMHO having a wait of 2 seconds is better than 1.5 seconds to avoid a too-fast unit test.
```suggestion
					time.Sleep(time.Millisecond * 2000)
```